openadapt/crud.py

```diff
@@ -221,6 +218,27 @@ def filter_stop_sequences(action_events):
         for _ in range(0, num_to_remove):
             action_events.pop()
 
+def save_screenshot_diff(screenshots):
+    data_updated = False
+    logger.info("precomputing diffs for screenshots...")
+
+    for screenshot in screenshots:
+        if not screenshot.prev:
+            continue
+        if not screenshot.png_diff_data:
+            screenshot.png_diff_data = screenshot.convert_png_to_binary(screenshot.diff)
+            data_updated = True
+        if not screenshot.png_diff_mask_data:
+            screenshot.png_diff_mask_data = screenshot.convert_png_to_binary(screenshot.diff_mask)
+            data_updated = True
+
+    if data_updated:
+        logger.info("saving screenshot diff data to db...")
+        db.bulk_save_objects(screenshots)
+        db.commit()
+
+    return screenshots
+
 
 def get_screenshots(recording, precompute_diffs=False):
     screenshots = _get(Screenshot, recording.timestamp)
```

Comment:
This feature should be activated via an environment variable flag for convenience. By default it should be deactivated to save CPU resources.

WDYT ?

Reply:
@0dm This is implemented now in https://github.com/OpenAdaptAI/OpenAdapt/pull/367/commits/cb14ec36c4b60afa8dfda412117f36b77a1761b5

Reply:
Also, I just updated the PR description to facilitate manual testing the feature.

Reply:
@0dm Any updates on my requests ?

Reply:
oops, sorry didn't see. lgtm